internal/corazawaf/rule.go

```diff
@@ -376,7 +376,7 @@ func (r *Rule) doEvaluate(logger debuglog.Logger, phase types.RulePhase, tx *Tra
 				// Flow actions are evaluated also if the rule engine is set to DetectionOnly
 				logger.Debug().Str("action", a.Name).Int("phase", int(phase)).Msg("Evaluating flow action for rule")
 				a.Function.Evaluate(r, tx)
-			} else if a.Function.Type() == plugintypes.ActionTypeDisruptive && tx.RuleEngine == types.RuleEngineOn {
+			} else if a.Function.Type() == plugintypes.ActionTypeDisruptive && tx.RuleEngine != types.RuleEngineOff {
```

Comment:
Thanks for the PR!
I think this is interesting, but something we should be quite careful with. It tweaks the core definition of `DetectionOnly`, which currently means that rules are processed but disruptive actions are never executed.
If I'm getting this right, with this PR, the logic becomes: disruptive actions are executed, but no interruption is triggered.

I see value in this change, it closes the gap between how the system behaves in `DetectionOnly`/`On` mode ( See https://github.com/corazawaf/coraza/issues/1051), but we should consider all the implications, as this changes the engine's behavior in DetectionOnly mode.

I'll take a more in-depth look